draft-ietf-rats-architecture.md

```diff
@@ -476,6 +476,14 @@ to the outside Verifier.
 
 In this situation, the trust model described in {{trustmodel}} is also suitable for this inside Verifier.
 
+## Roles and Entities
+
+An entity in the RATS architecture includes at least on of the roles defined in this document. As a result, the entity can participate as a constituent of the RATS architecture. Additionally, an entity can aggregate more than one role into itself. These collapsed roles combine the duties of multiple roles. In these cases, interaction between these roles do not necessarily use the Internet Protocol. They can, e.g. using a loopback device or other IP-based communication between separate environments, but they do not have to. Alternative channels to convey conceptual messages include sockets, GPIO interfaces, local busses, or hypervisors. This conveyance is sometime referred to as "local conveyance". This type of conveyance is typically found in Composite Devices. Most importantly, these conveyance methods are out-of-scope of RATS, but they are presumed to exist in order to convey conceptual messages appropriately between roles.
+
+An exemplary usage scenario: An entity that both connects to a wide-area network and to a system bus taking on both the Attester and Verifier roles. As a system bus entity, a Verifier consumes Evidence from other devices connected to the system bus that implement Attester roles. As a wide-area network connected entity, it may implement an Attester role. The entity, as a system bus Verifier, may choose to fully isolate its role as a wide-area network Attester. Alternatively, attestation results might reflect an aggregation of system bus operational state claims in the form of Evidence.
```

Comment:
I'm experiencing cognitive dissonance trying to relate the roles to both a bus connected example and a network connected example. It seems to be focusing on the composite device but highlighting the case where composite attesters could be network connected. However, I'm not certain based on this text. But otherwise, why wouldn't that discussion belong in the section on composite device?

I think the goal is to illustrate use cases where roles can be combined. Maybe sticking to a few intuitive examples is best. e.g., verifier + relying party, endorser + verifier and maybe verifier+owner. Combining attester and verifier is problematic because the trust model objective is to separate attester from verifier.

Reply:
TBH, I just copied your description from #73 into here. I probably set the wrong context?

Reply:
I assume line 401 is okay?

Reply:
Do you mean line 481?
I think everything after "interaction between these roles do not necessarily use the Internet Protocol" is not about role interactions but entity interactions. I don't know if entity interactions belongs or if it does should be a separate section or at least separate paragraph. The main point of line/paragraph 481 is that an entity can host (combine) multiple roles. When that happens, the role messages between roles on peer entities remains true to the roles architecture. For example, if Verifier and RP are combined. Evidence still flows to Verifier and RP still consumes Appraisal policy for Attestation Results (even though there isn't an Attestation Results flow exposed externally to the entity).

Reply:
It remains an interaction between roles, the means are changing though. Even if all roles would be collapsed, they'd still interact, "locally". Nothing changes architecturally, I think. I consequence, this is not intended to be about entity interaction. Maybe we should simply spell that out?